### PR TITLE
[SPARK-43498][PS][TESTS] Enable `StatsTests.test_axis_on_dataframe` for pandas 2.0.0.

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to enable `StatsTests.test_axis_on_dataframe` for pandas 2.0.0.

### Why are the changes needed?

Increase the test coverage.

### Does this PR introduce _any_ user-facing change?

No, it's test-only

### How was this patch tested?

Updated the existing UT.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: python/pyspark/pandas/tests/test_stats.py [hunk 1]
```diff
@@ -180,6 +176,10 @@ def test_axis_on_dataframe(self):
                 },
                 index=range(10, 15001, 10),
             )
+            # There is a regression in Pandas 2.1.0,
+            # so we should manually cast to float until the regression is fixed.
+            # See https://github.com/pandas-dev/pandas/issues/55194.
+            pdf = pdf.astype(float)
```
Comment (dongjoon-hyun) [inline python/pyspark/pandas/tests/test_stats.py:182]: If this is a `Pandas` bug, it seems that we need another TODO Spark JIRA ID to clean up this after Pandas community fixes it.

Comment (itholic) [reply python/pyspark/pandas/tests/test_stats.py:183]: Sounds good! Just created https://issues.apache.org/jira/browse/SPARK-45228 for tracking, and added comments to the test.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by dongjoon-hyun** [comment]
+1, LGTM, too.

**Pull request closed without merging**